src/stores/WalletStore.ts

```diff
@@ -38,24 +37,95 @@ class WalletStore extends PersistableStore {
     return web3Modal.cachedProvider
   }
 
-  async connect(clearCachedProvider = false) {
+  private async getUserNetworkName() {
+    return (await provider.getNetwork()).name
+  }
+
+  private async isNetworkRight() {
+    const network = env.VITE_ETH_NETWORK
+    const userNetwork = await this.getUserNetworkName()
+    return network === userNetwork
+  }
+
+  private async handleAccountChanged() {
+    if (!provider) return
+
+    this.walletLoading = true
+    this.account = await this.getAccount()
+    NotificationsStore.showTwitterShare = false
+    this.walletLoading = false
+  }
+
+  private async getAccount() {
+    return (await provider.listAccounts())[0]
+  }
+
+  private clearData() {
+    NotificationsStore.showTwitterShare = false
+    web3Modal.clearCachedProvider()
+    this.account = undefined
+  }
+
+  private async switchNetwork() {
+    const chainId = hexValue(env.VITE_CHAIN_ID)
+    await provider.send('wallet_switchEthereumChain', [{ chainId }])
+  }
+
+  private subscribeProvider(provider: Web3Provider) {
+    if (!provider.on) return
+
+    provider.on('error', (error: Error) => {
+      handleError(error)
+    })
+
+    provider.on('accountsChanged', (accounts: string[]) => {
+      if (!accounts.length) this.clearData()
+
+      this.account = undefined
+      void this.handleAccountChanged()
+    })
+    provider.on('disconnect', (error: unknown) => {
```

Comment:
btw, are you sure that `error` here is `unknown`, there is a `provider.on(error: Error)`, error has `Error` type, maybe here too?

Reply:
you're right, fixed